src/tools/perf/cuda/cuda_kernel.cu

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2025. ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
+
+#include <tools/perf/lib/libperf_int.h>
+#include <cuda_runtime.h>
+
+
+class cuda_ucp_test_runner: public ucp_perf_test_runner_base<uint64_t> {
+public:
+    using psn_t = uint64_t;
+
+    ucp_perf_test_device_runner(ucx_perf_context_t &perf) :
+        ucp_perf_test_runner_base<uint64_t>(perf)
+    {
+    }
+
+    ucs_status_t run()
+    {
+        switch (m_perf.params.test_type) {
+        case UCX_PERF_TEST_TYPE_PINGPONG:
+            switch (m_perf.params.command) {
+            case UCX_PERF_CMD_PUT_MULTI:
+                return run_pingpong();
+            default:
+                return UCS_ERR_INVALID_PARAM;
+            }
+        case UCX_PERF_TEST_TYPE_STREAM_UNI:
+            switch (m_perf.params.command) {
+            case UCX_PERF_CMD_PUT_MULTI:
+                return run_stream_uni();
+            default:
+                return UCS_ERR_INVALID_PARAM;
+            }
+        default:
+            return UCS_ERR_INVALID_PARAM;
+        }
+    }
+
+private:
+    ucs_status_t run_pingpong()
+    {
+        return UCS_OK;
+    }
+
+    ucs_status_t run_stream_uni()
+    {
+        size_t length = ucx_perf_get_message_size(&m_perf.params);
+        ucs_assert(length >= sizeof(psn_t));
+
+        m_perf.send_allocator->memset(m_perf.send_buffer, 0, length);
+        m_perf.recv_allocator->memset(m_perf.recv_buffer, 0, length);
+
+        ucp_perf_barrier(&m_perf);
+
+
+        ucx_perf_get_time(&m_perf);
+        ucp_perf_barrier(&m_perf);
+        return status;
+    }
+};
+
+static ucs_status_t
+ucp_perf_cuda_dispatch(ucx_perf_context_t *perf, ucx_perf_cmd_t cmd,
+                       ucx_perf_test_type_t type, unsigned flags)
+{
+    return UCS_OK;
+}
+
+UCS_STATIC_INIT {
+    static ucx_perf_device_dispatcher_t cuda_dispatcher = {
+        .ucp_dispatch = ucp_perf_cuda_dispatch,
+    };
+
+    ucx_perf_mem_type_device_runners[UCS_MEMORY_TYPE_CUDA]         = &cuda_dispatcher;
+    ucx_perf_mem_type_device_runners[UCS_MEMORY_TYPE_CUDA_MANAGED] = &cuda_dispatcher;
```

Comment:
i don't think we need it per mem type. because mem type is also part of the perf test structure. maybe just a global array/list of available dispatchers.